app/controllers/concerns/authenticable.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+module Authenticable
+  extend ActiveSupport::Concern
+
+  included do
+    before_action :authenticate_user_using_x_auth_token
+    before_action :authenticate_user!
+  end
+
+  private
+
+    def authenticate_user_using_x_auth_token
+      user_email = request.headers["X-Auth-Email"]
+      auth_token = request.headers["X-Auth-Token"].presence
+      user = user_email && User.find_by_email(user_email)
```

Comment:
Don't use dynamic finder methods. Make changes wherever necessary.

https://www.bigbinary.com/learn-rubyonrails-book/find-vs-find-by-vs-where-in-active-record#don-t-use-dynamic-finder-methods